__tests__/app.test.js

```diff
@@ -258,6 +258,81 @@ describe('GET /api/articles/:article_id/comments', () => {
         });
     });
 })
+describe('POST /api/articles/:article_id/comments', () => {
+    test('should respond with a status code of 201 and the posted comment ', () => {
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(201)
+        .send({ username: 'butter_bridge', body: 'Thought provoking read!' })
+        .then(({body: { comment }}) =>{
+                expect(comment).toEqual(expect.objectContaining({
+                    article_id: 1,
+                    comment_id: expect.any(Number),
+                    votes: 0,  
+                    created_at: expect.any(String), //There is a utility class tranlating a time stamp into date
+                    author: 'butter_bridge',
+                    body: 'Thought provoking read!',
+                }))
+            
+        })
+        
+    });
+    test('should respond with a status code of 400 when given a bad article_id', () => {
+        return request(app)
+        .post('/api/articles/something/comments')
+        .expect(400)
+        .send({ username: 'butter_bridge', body: 'Thought provoking read!' })
+        .then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad Request");
+          });
+    })
+    test('should respond with a status code of 400 when given a article_id that doesnt exist', () => {
+        return request(app)
+        .post('/api/articles/1000000/comments')
+        .expect(400)// this has previously in others been a 404 but in this one it seems like it should be a 400 because its attempting to add something that cant be added
+        .send({ username: 'butter_bridge', body: 'Thought provoking read!' })
+        .then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad Request");
+          });
+    })
+    test('should respond with a status code of 400 when given a username that doesnt exist', () => {
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(400)
+        .send({ username: 'someone', body: 'Thought provoking read!' })
+        .then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad Request");
+          });
+    })
+    test('should respond with a status code of 400 when given a object with too many properties', () => {
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(400)
+        .send({ username: 'butter_bridge', body: 'Thought provoking read!', somethingelse: 234 })
+        .then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad Request");
+          });
+    })
+    test('should respond with a status code of 400 when given a object with too few properties', () => {
```

Comment:
nice idea. I think a better description would be "missing *required* fields" because you may find that in the future some made be optional and depending how you handle this in your code it might not scale very well